lib/node_modules/@stdlib/array/float16/lib/polyfill.js

```diff
@@ -0,0 +1,903 @@
+* @returns {Float16Array} New Float16Array with filtered elements
+*/
+Float16Array.prototype.filter = function( callback, thisArg ) {
+    var i;
+    var value;
+    var result;
+
+    if ( typeof callback !== 'function' ) {
+        throw new TypeError( 'invalid argument. Must provide a function. Value: `' + callback + '`.' );
+    }
+
+    result = [];
+    for ( i = 0; i < this.length; i++ ) {
+        value = this.get( i );
+        if ( callback.call( thisArg, value, i, this ) ) {
+            result.push( value );
+        }
+    }
+    return new Float16Array( result );
+};
+
+/**
+* Returns the value of the first element that satisfies the provided testing function.
+*
+* @param {Function} callback - Function to execute on each value
+* @param {*} [thisArg] - Value to use as `this` when executing `callback`
+* @throws {TypeError} Must provide a function as callback
+* @returns {*} First matching element, or undefined if no match
+*/
+Float16Array.prototype.find = function( callback, thisArg ) {
+    var i;
+    var value;
+
+    if ( typeof callback !== 'function' ) {
+        throw new TypeError( 'invalid argument. Must provide a function. Value: `' + callback + '`.' );
+    }
+
+    for ( i = 0; i < this.length; i++ ) {
+        value = this.get( i );
+        if ( callback.call( thisArg, value, i, this ) ) {
+            return value;
+        }
+    }
+    return void 0; // returns undefined
+};
+
+/**
+* Returns the index of the first element that satisfies the provided testing function.
+*
+* @param {Function} callback - Function to execute on each value
+* @param {*} [thisArg] - Value to use as `this` when executing `callback`
+* @throws {TypeError} Must provide a function as callback
+* @returns {integer} Index of found element, or -1 if none found
+*/
+Float16Array.prototype.findIndex = function( callback, thisArg ) {
+    var i;
+    var value;
+
+    if ( typeof callback !== 'function' ) {
+        throw new TypeError( 'invalid argument. Must provide a function. Value: `' + callback + '`.' );
+    }
+    for ( i = 0; i < this.length; i++ ) {
+        value = this.get( i );
+        if ( callback.call( thisArg, value, i, this ) ) {
+            return i;
+        }
+    }
+    return -1;
+};
+
+/**
+* Return an iterator over [index, value] pairs.
+*
+* @returns {Object} Iterator
+*/
+Float16Array.prototype.entries = function() {
+    var self = this;
+    var i = 0;
+
+    return {
+        next: function() {
+            if ( i < self.length ) {
+                var v = [ i, self.get( i ) ];
+                i += 1;
+                return {
+                    value: v,
+                    done: false
+                };
+            }
+            return {
+                value: void 0,
+                done: true
+            };
+        }
+    };
+};
+
+/**
+* Return an iterator over indices.
+*
+* @returns {Object} Iterator
+*/
+Float16Array.prototype.keys = function() {
```

Comment:
We don't use anonymous functions. We used named functions to assist in making stack traces more legible.